tests/test_config.py

```diff
@@ -1376,11 +1376,16 @@ def test_factors(self, newconfig):
                 a: dep-a
                 b: dep-b
                 x: dep-x
+                !a: dep-!a
+                !b: dep-!b
+                !x: dep-!x
         """
         conf = newconfig([], inisource)
         configs = conf.envconfigs
-        assert [dep.name for dep in configs['a-x'].deps] == ["dep-all", "dep-a", "dep-x"]
-        assert [dep.name for dep in configs['b'].deps] == ["dep-all", "dep-b"]
+        assert [dep.name for dep in configs['a-x'].deps] == \
```

Comment:
I prefer avoiding using the line separator - instead consider setting out the right hand side to a local variable e.g. ``exp_deps``.

Reply:
ok, changing it to:
```
        expected = ["dep-all", "dep-a", "dep-x", "dep-!b"]
        assert [dep.name for dep in configs['a-x'].deps] == expected
        expected = ["dep-all", "dep-b", "dep-!a", "dep-!x"]
        assert [dep.name for dep in configs['b'].deps] == expected
```
Though I'd personally prefer:
```
        assert [dep.name for dep in configs['a-x'].deps] == [
            "dep-all", "dep-a", "dep-x", "dep-!b"]
        assert [dep.name for dep in configs['b'].deps] == [
            "dep-all", "dep-b", "dep-!a", "dep-!x"]
```